src/asl.ts

```diff
@@ -1255,7 +1255,9 @@ export class ASL {
 
     if (isPromiseExpr(expr)) {
       // if we find a promise, ensure it is wrapped in Await or returned then unwrap it
-      if (isAwaitExpr(expr.parent) || isReturnStmt(expr.parent)) {
+      if (
+        expr.hasOnlyAncestors(isConditionExpr, anyOf(isReturnStmt, isAwaitExpr))
```

Comment:
Does it stop at the first one it finds?

Reply:
The first predicate is a passthrough and the second it stops at